pandas/core/arrays/boolean.py

```diff
@@ -565,33 +568,38 @@ def logical_method(self, other):
                 # Rely on pandas to unbox and dispatch to us.
                 return NotImplemented
 
+            assert op.__name__ in {"or_", "ror_", "and_", "rand_", "xor", "rxor"}
             other = lib.item_from_zerodim(other)
+            other_is_booleanarray = isinstance(other, BooleanArray)
+            other_is_scalar = lib.is_scalar(other)
             mask = None
 
-            if isinstance(other, BooleanArray):
+            if other_is_booleanarray:
                 other, mask = other._data, other._mask
             elif is_list_like(other):
                 other = np.asarray(other, dtype="bool")
                 if other.ndim > 1:
                     raise NotImplementedError(
                         "can only perform ops with 1-d structures"
                     )
-                if len(self) != len(other):
-                    raise ValueError("Lengths must match to compare")
                 other, mask = coerce_to_array(other, copy=False)
 
-            # numpy will show a DeprecationWarning on invalid elementwise
-            # comparisons, this will raise in the future
-            with warnings.catch_warnings():
-                warnings.filterwarnings("ignore", "elementwise", FutureWarning)
-                with np.errstate(all="ignore"):
-                    result = op(self._data, other)
+            if other_is_scalar and not isinstance(other, (type(libmissing.NA), bool)):
```

Comment:
You'll need to add `np.bool_` as well I think (and tests for this), as a numpy scalar is not an instance of a bool:

```
In [14]: isinstance(np.bool_(True), bool) 
Out[14]: False
```

Reply:
Also, `other is pd.NA or isinstance(other, bool)` is a bit faster, although I am not sure we care about the 60 ns difference

Reply:
Hmm, but I see that you actually already have a test for numpy scalars?

Reply:
Ah, I was tripped up by `np.bool` vs `np.bool_`.